x-pack/plugins/stack_connectors/public/connector_types/es_index/es_index_params.tsx

```diff
@@ -77,15 +82,19 @@ export const IndexParamsFields = ({
     // eslint-disable-next-line react-hooks/exhaustive-deps
   }, [actionConnector?.id]);
 
-  const onDocumentsChange = (updatedDocuments: string) => {
+  const onDocumentsChange = (updatedDocuments: string | null) => {
     try {
-      const documentsJSON = JSON.parse(updatedDocuments);
-      editAction('documents', [documentsJSON], index);
+      if (updatedDocuments != null) {
+        const documentsJSON = JSON.parse(updatedDocuments);
+        editAction('documents', [documentsJSON], index);
+      } else {
+        editAction('documents', updatedDocuments, index);
+      }
       setDocumentToIndex(updatedDocuments);
     } catch (e) {
       // set document as empty to turn on the validation for non empty valid JSON object
       editAction('documents', [{}], index);
-      setDocumentToIndex(undefined);
+      setDocumentToIndex(updatedDocuments);
```

Comment:
I think I'm missing why we want to keep `updatedDocuments`. Is it because we want to keep showing in the invalid json but still have an error shown?

Reply:
You are right and that change was unnecessary so I reverted it.